wagtail/images/models.py

```diff
@@ -651,17 +651,22 @@ def create_renditions(
         return_value: Dict[Filter, AbstractRendition] = {}
         filter_map: Dict[str, Filter] = {f.spec: f for f in filters}
 
-        with self.open_file() as file:
-            original_image_bytes = file.read()
+        _, tempfile = mkstemp()
+
+        with open(tempfile, "wb") as f:
+            with self.open_file() as file:
+                copyfileobj(file, f)
 
         to_create = []
 
         def _generate_single_rendition(filter):
             # Using ContentFile here ensures we generate all renditions. Simply
             # passing self.file required several page reloads to generate all
-            image_file = self.generate_rendition_file(
-                filter, source=ContentFile(original_image_bytes, name=self.file.name)
-            )
+            with open(tempfile, "rb") as f:
```

Comment:
note to self: check whether there's a risk of a race condition

Reply:
Because we only ever read `tempfile`, and create new file handles to it with each read, there shouldn't be any chance of a race. The file will be flushed and closed by the parent thread long before a child thread gets to it too, so the file will always contain all the data.

Reply:
That is my understanding of the process too.
Did some check locally and it checks out.